components/eam/src/chemistry/mozart/mo_chm_diags.F90

```diff
@@ -448,19 +455,46 @@ subroutine chm_diags( lchnk, ncol, vmr, mmr, rxt_rates, invariants, depvel, depf
 
     do k = 1,pver
        mass(:ncol,k) = pdel(:ncol,k) * area(:ncol) * rgrav
+       drymass(:ncol,k) = pdeldry(:ncol,k) * area(:ncol) * rgrav
     enddo
 
     call outfld( 'AREA', area(:ncol),   ncol, lchnk )
     call outfld( 'MASS', mass(:ncol,:), ncol, lchnk )
+    call outfld( 'DRYMASS', drymass(:ncol,:), ncol, lchnk )
 
-    ! convert ozone from mol/mol to DU
-    wrk(:ncol,:) = pdel(:ncol,:)*vmr(:ncol,:,id_o3)*avogadro*rgrav/air_molwgt/DUfac*1.e3_r8
-    ! total column ozone, vertical integration
+    ! convert ozone from mol/mol (w.r.t. dry air mass) to DU
+    wrk(:ncol,:) = pdeldry(:ncol,:)*vmr(:ncol,:,id_o3)*avogadro*rgrav/mwdry/DUfac*1.e3_r8
```

Comment:
Hi @tangq , Though no impact on the results, this conversion is repeated three times. I think it is preferable doing the conversion only once and putting them on wrk, then using a different horizontal-only work array for the calculations of TOZ, SCO and TCO.

Reply:
@wlin7 , thanks for the suggestions. I implemented the changes per your suggestion in the new commit.

Reply:
I also confirmed that the new commit gives the same results for ozone columns.